src/types/manager.ts

```diff
@@ -103,6 +103,7 @@ export type FinalFirmware = BaseFirmware & {
   mcu_versions: Array<Id>;
   application_versions: Array<Id>;
   bytes?: number;
+  updateAvailable?: FirmwareUpdateContext | null | undefined;
```

Comment:
As I said, `updateAvailable` is set by `getLatestFirmwareForDevice`, so it can return `null` depending on wether or not it finds a new version of BOLOS to update.

Concerning the difference between "set and undef" and "not set", there is no difference for us in our implementation specifically as far as I could tell, but you could get into trouble the day we need an `Object.keys(<FinalFirmware>)` which would then have different returns if the key is defined or not, regardless of its value being defined. I guess that's the use case Flow is trying to prevent by differentiating the two approaches.